### PR TITLE
Feature: Add CMC integration

### PR DESCRIPTION
**Pull request by chumaknadya**

### CODE REVIEW

File: app/api/v2/coin_market_cap/assets.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module API
+  module V2
+    module CoinMarketCap
+      class Assets < Grape::API
+        desc 'Details on crypto currencies available on the exchange'
+        get '/assets' do
+          # Clarify: Visible or all ?
+          Currency.coins.ordered.map do |currency|
```
Comment (mnaichuk) [inline app/api/v2/coin_market_cap/assets.rb]: we can use redis here

File: app/api/v2/coin_market_cap/summary.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module API
+  module V2
+    module CoinMarketCap
+      class Summary < Grape::API
+        desc 'Overview of market data for all tickers and all market pairs on the exchange'
+        get '/summary' do
+          # Clarify: visible or all?
+          ::Market.ordered.map do |market|
```
Comment (mnaichuk) [inline app/api/v2/coin_market_cap/summary.rb]: we can use redis here

File: app/api/v2/coin_market_cap/ticker.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module API
+  module V2
+    module CoinMarketCap
+      class Ticker < Grape::API
+        desc 'Get 24-hour pricing and volume summary for each market pair'
+        get '/ticker' do
+          # Clarify: visible or all?
```
Comment (mnaichuk) [inline app/api/v2/coin_market_cap/ticker.rb]: we can use redis here

**Review by mnaichuk — CHANGES_REQUESTED** [review]

File: spec/api/v2/coin_market_cap/assets_spec.rb [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+describe API::V2::CoinMarketCap::Assets, type: :request do
+  describe 'GET /api/v2/coinmarketcap/assets' do
+    before(:each) { clear_redis }
+
+    context 'There are currencies' do
+      context 'with unified id' do
+        before do
+          Currency.coins.each do |currency|
+            stub_request(:get, "http://pro-api.coinmarketcap.com/v1/cryptocurrency/map?CMC_PRO_API_KEY=UNIFIED-CRYPTOASSET-INDEX&"\
```
Comment (calj) [inline spec/api/v2/coin_market_cap/assets_spec.rb]: can we use https in every request to CMC?

File: app/api/v2/coin_market_cap/helpers.rb [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          }
+        end
+
+        def format_trade(trade)
+          {
+            trade_id:     trade[:id],                       # mandatory [integer]: A unique ID associated with the trade for the currency pair transaction
+            price:        trade[:price],                    # mandatory [decimal]: Last transacted price of base currency based on given quote currency
+            base_volume:  trade[:amount],                   # mandatory [decimal]: Transaction amount in BASE currency
+            quote_volume: trade[:total],                    # mandatory [decimal]: Transaction amount in QUOTE currency
+            timestamp:    DateTime.now.strftime('%Q').to_i, # mandatory [integer]: Unix timestamp in milliseconds for when the transaction occurred
```
Comment (mnaichuk) [inline app/api/v2/coin_market_cap/helpers.rb]: Trade time

**Review by mnaichuk — APPROVED** [review]

**Pull request merged**